website/docs/basics/dynamic-import.md

```diff
@@ -0,0 +1,130 @@
+* reduce the size of finished bundled JavaScript code.
+* have websites that are fast and this can positively impact your Seo rating.
+
+## How does it work?
+
+In a nutshell, you load all your code using JavaScript promises and Webpack splits all these promises into separate file chinks. Once the condition to resolve the promise is set (generally we check if the selector is present in the DOM), Webpack automatically injects your chink in the DOM and loads it up. This is why you will see in your public folder a bunch of smaller files called (0.js, 1.js, 2.js, etc).
+
+You don't need to think about anything here because Webpack does it for you.
+
+## Example
+
+For example, let's make a carousel. We like to write all our JavaScript code in classes so we will provide an example for that but you can use this with functions as well.
+
+**We have to files:**
+* **index.js** - the main entry point for your JavaScript feature.
+* **className.js** - class with your JavaScript features that you can reuse.
+
+
+### index.js
+
+This is a version if you can have multiple instances of this feature on one page. Generally, if you are creating a feature for your block/component you will use this version.
+
+```js
+import domReady from '@wordpress/dom-ready';
+
+// Use this helper to make sure this code is executed when everything in DOM is set.
+domReady(() => {
+
+  // Provide selectors to check.
+  const selector = '.js-block-carousel';
+  const elements = document.querySelectorAll(selector);
+
+  // This is the important part because if this condition is true this promise will resolve and your chink will be loaded in the DOM.
+  if (elements.length) {
+
+    // Normally load and resolve a promise on file import.
+    import('./carousel-slider').then(({ CarouselSlider }) => {
+
+      // Loop possible multiple iterations.
+      [...elements].forEach((element) => {
+
+        // Load a class for each iteration.
+        const carouselSlider = new CarouselSlider({
+          element,
+          nextElement: `${selector}-next-arrow`,
+          prevElement: `${selector}-prev-arrow`,
+        });
+
+        // Run and init function that starts everything in your class.
+        carouselSlider.init();
+      });
+    });
+  }
+});
+```
+
+If you are sure you will have only one instance of this feature on one-page use this code in your index.js file.
+
+```js
+import domReady from '@wordpress/dom-ready';
+
+// Use this helper to make sure this code is executed when everything in DOM is set.
+domReady(() => {
+
+  // Provide selectors to check.
+  const selector = '.js-block-carousel';
+  const element = document.querySelector(selector);
+
+  // This is the important part because if this condition is true this promise will resolve and your chink will be loaded in the DOM.
+  if (element) {
+
+    // Normally load and resolve a promise on file import.
+    import('./carousel-slider').then(({ CarouselSlider }) => {
+      // Load a class for each iteration.
```

Comment:
```suggestion
      // Load a class for all sliders
```

Because (at least specifically for the carousel in this example) it's doesn't initialize the class for each slider on the page but rather (I assume) it initializes all of them in the `init()` method. (If not, we need a look here for each Carousel on the page)